openslides_backend/action/actions/poll/mixins.py

```diff
@@ -74,22 +76,28 @@ def on_stop(self, instance: Dict[str, Any]) -> None:
 
         # stop poll in vote service and create vote objects
         results = self.vote_service.stop(instance["id"])
+        votes_raw = results["votes"]
+        votes = json.loads(votes_raw)
+        if poll["type"] == Poll.TYPE_CRYPTOGRAPHIC:
+            votes = votes["votes"]
+            instance["votes_raw"] = votes_raw
+            instance["votes_signature"] = results.get("signature")
         action_data = []
         votesvalid = Decimal("0.000000")
         option_results: Dict[int, Dict[str, Decimal]] = defaultdict(
             lambda: defaultdict(lambda: Decimal("0.000000"))
         )  # maps options to their respective YNA sums
-        for ballot in results["votes"]:
+        for ballot in votes:
             user_token = get_user_token()
-            vote_weight = Decimal(ballot["weight"])
+            vote_weight = Decimal(ballot.get("weight")) or Decimal("1.000000")
```

Comment:
This does not work if the attribute "weight" is not in ballot. This is the case for cryptographic votes. The following code works:

```python
vote_weight = Decimal("1.000000")
if ballot.get("weight",None) is not None:
    vote_weight = Decimal(ballot.get("weight"))
```

Reply:
done